File: src/map_publisher.cpp
#include "ros/ros.h"

#include "std_msgs/Float32.h"
#include "std_msgs/Float32MultiArray.h"
#include "sensor_msgs/Joy.h"
#include "sensor_msgs/JointState.h"
#include <tf/transform_broadcaster.h>

#include <visualization_msgs/Marker.h>

#include "math.h"
#include <sstream>
#include <string>

#define MAP_SIZE 15
float map_points[MAP_SIZE][2]={
	{0.0,0.0},
	{0.5,0.0},
	{1.0,0.0},
	{1.5,0.0},
	{2.0,0.0},

	{2.5,0.1},
	{2.9,0.5},
	{3.0,1.0},
	{2.9,1.5},
	{2.5,1.9},
	
	{2.0,2.0},
	{1.5,2.0},
	{1.0,2.0},
	{0.5,2.0},
	{0.0,2.0},
};



int main(int argc, char **argv)
{
	ros::init(argc, argv, "map_publisher");
	ros::NodeHandle n;
	ros::NodeHandle pn("~");
	
	//publish
	ros::Publisher map_pub = n.advertise<std_msgs::Float32MultiArray>("map_data", 10,true);
	
	std_msgs::Float32MultiArray map_data;
	
	for(int i=0;i<MAP_SIZE;i++){
		map_data.data.push_back(map_points[i][0]);
		map_data.data.push_back(map_points[i][1]);
	}

	map_pub.publish(map_data);

	ros::Rate loop_rate(10); 
	while (ros::ok()){
		ros::spinOnce();
		loop_rate.sleep();
	} 
 	return 0;
}

File: scripts/map_publisher.py
#!/usr/bin/env python
import rospy
from std_msgs.msg import Float32
from std_msgs.msg import Float32MultiArray
from visualization_msgs.msg import Marker
from geometry_msgs.msg import Point
import getWayPoints

array = []
map_data = Float32MultiArray(data=array)

array = []
bl_tube_data = Float32MultiArray(data=array)

array = []
br_tube_data = Float32MultiArray(data=array)


rospy.init_node("map_publisher")
map_pub = rospy.Publisher("map_data", Float32MultiArray, queue_size=10)
bl_tube_pub = rospy.Publisher("bl_tube_data", Float32MultiArray, queue_size=10)
br_tube_pub = rospy.Publisher("br_tube_data", Float32MultiArray, queue_size=10)

rate = rospy.Rate(10)

map_points = getWayPoints.read_rob()
bl_tube_points = getWayPoints.read_bl_tube_points()
br_tube_points = getWayPoints.read_br_tube_points()

for i in range(len(map_points)):
    map_data.data.append(map_points[i][0])
    map_data.data.append(map_points[i][1])

for i in range(len(bl_tube_points)):
    map_data.data.append(bl_tube_points[i][0])
    map_data.data.append(bl_tube_points[i][1])

for i in range(len(br_tube_points)):
    map_data.data.append(br_tube_points[i][0])
    map_data.data.append(br_tube_points[i][1])



while not rospy.is_shutdown():
    map_pub.publish(map_data)
    bl_tube_pub.publish(bl_tube_data)
    br_tube_pub.publish(br_tube_data)
    rate.sleep()
